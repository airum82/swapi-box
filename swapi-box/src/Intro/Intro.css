.intro {
  background-color: black;
  color: yellow;
  font-size: 2em;
  height: 100%; }

@-webkit-keyframes scroll {
  0% {
    -webkit-transform: translate(0, 100%); }
  100% {
    -webkit-transform: translate(0, -100%); } }

.slide-up {
  overflow: hidden; }

.slide-up div {
  display: inline-block;
  -webkit-animation: scroll 25s infinite linear; }
