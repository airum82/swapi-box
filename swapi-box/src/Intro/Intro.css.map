{"version":3,"sourceRoot":"","sources":["Intro.scss"],"names":[],"mappings":"AAAA;EACE","file":"Intro.css"}