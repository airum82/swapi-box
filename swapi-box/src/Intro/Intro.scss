.intro {
  background-color: black;
  color: yellow;
  font-size: 2em;
  height: 100vh; 
}

@-webkit-keyframes scroll {
  0% {
      -webkit-transform: translate(0, 100%);

  }

  100% {
      -webkit-transform: translate(0, -100%)

  }
}

.slide-up {
  overflow: hidden;
}

.slide-up div {
  display: inline-block;
  -webkit-animation: scroll 20s infinite linear;
}
