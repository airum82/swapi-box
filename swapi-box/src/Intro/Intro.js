import React from 'react';
import './Intro.css';

export const Intro = ({ intro }) => {
  return (
    <aside className="intro slide-up">
      <div>
        <p>{intro.intro}</p>
        <h3>{intro.title}</h3>
        <h5>{intro.releaseDate}</h5>
      </div>
    </aside>
  )
}