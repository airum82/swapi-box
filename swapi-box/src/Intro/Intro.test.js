import React from 'react';
import { Intro } from './Intro';
import { shallow } from 'enzyme';

describe('Intro component', () => {
  it('should match the snapshot upon render', () => {
    const wrapper = shallow(<Intro 
                              intro={{}}
                            />);
    expect(wrapper).toMatchSnapshot();
  })
})