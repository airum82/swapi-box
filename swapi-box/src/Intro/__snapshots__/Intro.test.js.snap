// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component should match the snapshot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    intro={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <p />
        <h3 />
        <h5 />
</div>,
      "className": "intro slide-up",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p />,
          <h3 />,
          <h5 />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
      ],
      "type": "div",
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <p />
          <h3 />
          <h5 />
</div>,
        "className": "intro slide-up",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p />,
            <h3 />,
            <h5 />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h5",
          },
        ],
        "type": "div",
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
